src/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -25,15 +27,24 @@ public RoslynRequestExecutionQueue(AbstractLanguageServer<RequestContext> langua
             _initializeManager = languageServer.GetLspServices().GetRequiredService<IInitializeManager>();
         }
 
-        public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool rethrowExceptions)
+        public override async Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool rethrowExceptions)
         {
             if (rethrowExceptions)
             {
-                return nonMutatingRequestTask;
+                try
```

Comment:
fixes a bug here where we would only report NFW for non-mutating requests.  If a mutating request failed we would never get fault telemetry for it.